### PR TITLE
Adding migration for runc hook

### PR DESCRIPTION
**Pull request by sudermanjr**
Adds the hook if it does not exist.  Modifies it to newest if it is the old one.

### CODE REVIEW

**Comment by sudermanjr** [comment]
Strange, while testing this it does the updating of the hook properly in some places and not in others.

**Review by ejether — APPROVED** [review]
Migration itself looks good, but the file name needs include the starting version, not the new version.
So `migration_3_1_0.py` -> `migration_2_7_2.py`

**Comment by ejether** [comment]
I rebased your branch so it got the pipeline fixes that we were working on yesterday.

**Review by mjhuber — APPROVED** [review]
LGTM, thanks for updating.  This will come in handy

File: pentagon/migration/migrations/migration_2_7_2.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    # Setup cluster spec with patch-runc hook
+                    if True:
+                        cluster_spec_file = "{}/cluster-config/cluster.yml".format(item_path)
+                        with open(cluster_spec_file) as yaml_file:
+                            cluster_config = yaml.load(yaml_file.read())
+                            cluster_spec = cluster_config['spec']
+
+                            if cluster_spec.get('kubelet') is None:
+                                cluster_spec['kubelet'] = {}
+                            cluster_spec['kubelet']['anonymousAuth'] = False
```
Comment (davekonopka) [inline pentagon/migration/migrations/migration_2_7_2.py]: This change was already made in a previous migration. I'd remove it from this one.

**Review by davekonopka — APPROVED** [review]
Looks good to me. Minor comment left inline. I haven't tested out the migration fwiw.

File: pentagon/migration/migrations/migration_2_7_2.py (hunk 1 above)
Comment (sudermanjr) [reply pentagon/migration/migrations/migration_2_7_2.py]: Good catch.  Removed.

**Comment by sudermanjr** [comment]
@ejether  Do I need to change versions on this?

```
GIT master: Already up to date.
2019-03-11 09:54:22 root[67405]     INFO Pentagon Version: 3.1.0
2019-03-11 09:54:22 root[67405]     INFO Starting Repository Version: 2.7.3
2019-03-11 09:54:22 root[67405]     INFO No Migrations to run.
2019-03-11 09:54:22 root[67405]     INFO Installed Version > Repository Version.
2019-03-11 09:54:22 root[67405]     INFO  Use `pentagon migrate --yes` to update Repository
```

**Comment by ejether** [comment]
https://github.com/reactiveops/pentagon/blob/master/pentagon/migration/__init__.py#L60

Yes, I think I led you astray. The migration name should be `pentagon/migration/migrations/migration_2_7_3.py`. The migration finding check is picky like that.

**Comment by sudermanjr** [comment]
That did the trick.  If tests pass I'll ship this.

**Pull request merged**